plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -28,17 +42,59 @@
  */
 public class GoogleLoginService {
 
-  // TODO(chanseok): this will be needed later.
-  public static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
-      "https://www.googleapis.com/auth/userinfo#email"
+  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
+      "email", //$NON-NLS-1$
+      "https://www.googleapis.com/auth/cloud-platform" //$NON-NLS-1$
   ));
 
   /**
    * Returns the credential of the active user. If there is no active user, returns {@code null}.
    */
   // Should probably be synchronized properly.
-  // TODO(chanseok): consider returning a String JSON (i.e., hide Credential)
-  public Credential getActiveCredential() {
-    return null;
+  public Credential getActiveCredential(IShellProvider shellProvider) throws IOException {
+    return logIn(shellProvider);
+  }
+
+  private Credential logIn(IShellProvider shellProvider) throws IOException {
+    GoogleAuthorizationCodeRequestUrl requestUrl =
+        new GoogleAuthorizationCodeRequestUrl(Constants.getOAuthClientId(),
+                                              GoogleOAuthConstants.OOB_REDIRECT_URI,
+                                              OAUTH_SCOPES);
+
+    try {
+      IWorkbenchBrowserSupport browserSupport = PlatformUI.getWorkbench().getBrowserSupport();
+      browserSupport.getExternalBrowser().openURL(requestUrl.toURL());
+    } catch (PartInitException pie) {
+      MessageDialog.openError(shellProvider.getShell(),
+          Messages.ERROR_BROWSER_LAUNCHING, pie.getMessage());
+      return null;
+    }
+
+    InputDialog dialog = new InputDialog(shellProvider.getShell(),
+        Messages.ENTER_VERIFICATION_CODE_DIALOG_TITLE,
+        Messages.ENTER_VERIFICATION_CODE_DIALOG_MESSAGE, null, null);
+    if (dialog.open() != InputDialog.OK) {
+      return null;
+    }
+
+    String verificationCode = dialog.getValue();
+    GoogleAuthorizationCodeTokenRequest authRequest = new GoogleAuthorizationCodeTokenRequest(
+        Utils.getDefaultTransport(), Utils.getDefaultJsonFactory(),
+        Constants.getOAuthClientId(), Constants.getOAuthClientSecret(),
+        verificationCode,
+        GoogleOAuthConstants.OOB_REDIRECT_URI);
+
+    return createCredential(authRequest.execute());
```

Comment:
To enable @akerekes, you could stash the credential on the IWorkbench/MApplication. Something like:
```
MApplication app = PlatformUI.getWorkbench().getService(MApplication.class);
app.getTransientData().put("OAUTH_CRED", credential);
```
That value will be tossed on restart.

(Just in case: `MApplication` comes from an intermediate layer between the Workbench layer and the SWT widgets, called the _Modelled Workbench_.)